proxy/src/config.rs

```diff
@@ -79,6 +79,14 @@ pub struct Config {
     pub dns_max_ttl: Option<Duration>,
 }
 
+#[derive(Clone, Debug)]
+pub struct Namespaces {
```

Comment:
> I like that this is broken out into its own struct. This is definitely out of scope for this PR, but I wonder if we want to consider grouping other related configuration values into separate structs...

I just did it here because they need to be passed around a lot, and I didn't want to pass them as two `String` arguments or two `Namespace` arguments that could accidentally be swapped. I think if that kind of error-prone situation presents itself elsewhere, we should definitely solve it this way or a better way.